src/SonarScanner.MSBuild.Shim/SonarEngineWrapper.cs

```diff
@@ -84,9 +85,10 @@ private bool Execute(ProcessRunnerArguments args)
         return result.Succeeded;
     }
 
-    private static IEnumerable<ProcessRunnerArguments.Argument> JavaParams(AnalysisConfig config)
+    private IEnumerable<ProcessRunnerArguments.Argument> JavaParams(AnalysisConfig config, IAnalysisPropertyProvider userCmdLineArguments)
```

Comment:
Was this copied (in parts) from the scanner-cl wrapper? If so consider cross-linking the implementations with comments, so we have a chance of spotting any changes in one place might need to be applied in the other one as well.